src/main/java/com/es/phoneshop/web/PriceHistoryPageServlet.java

```diff
@@ -0,0 +1,37 @@
+package com.es.phoneshop.web;
+
+import com.es.phoneshop.model.product.ArrayListProductDao;
+import com.es.phoneshop.model.product.NullValuePassedException;
+import com.es.phoneshop.model.product.ProductDao;
+
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+public class PriceHistoryPageServlet extends HttpServlet {
+    private ProductDao productDao;
+
+    @Override
+    public void init(ServletConfig config) throws ServletException {
+        super.init(config);
+        productDao = ArrayListProductDao.getInstance();
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        String productId = request.getPathInfo();
+        try {
+            request.setAttribute("product", productDao.getProduct(Long.valueOf(productId.substring(1))));
+        } catch (NullValuePassedException e) {
+            throw new RuntimeException("Null value passed");
+        }
```

Comment:
Hiding exception cause here, is it intentional? I would better set it as cause for RuntimeException